src/main/java/Duke/TaskList.java

```diff
@@ -0,0 +1,214 @@
+     */
+    public ToDo makeToDoTask(String userInput) throws DukeException {
+        if (userInput.length() == "todo".length()) {
+            throw new DukeException("OOPS!!! The description of a event cannot be empty.");
+        }
+        return new ToDo(userInput.substring("todo".length() + 1));
+    }
+
+    /**
+     * Returns newly created Event task otherwise throw an appropriate error.
+     *
+     * @param userInput user input String
+     * @return Event task
+     * @throws DukeException exception thrown by Duke
+     */
+    public Task makeEventTask(String userInput) throws DukeException {
+        int indexOfAt = userInput.indexOf("/at");
+        int endIndexForEvent = "event".length();
+        boolean doesAtExist = indexOfAt != -1;
+        boolean doesTimeExist = userInput.length() > indexOfAt + "/at".length() + 1;
+        boolean doesEventDescriptionExist = indexOfAt != endIndexForEvent + 1;
+
+        if (!doesAtExist) {
+            throw new DukeException("OOPS!!! You are missing /at before your date");
+        }
+        if (!doesTimeExist) {
+            throw new DukeException("OOPS!!! The date of a event cannot be empty.");
+        }
+        if (!doesEventDescriptionExist) {
+            throw new DukeException("OOPS!!! The description of a event cannot be empty.");
+        }
+
+        return new Event(userInput.substring("event".length() + 1, indexOfAt - 1),
+                userInput.substring(indexOfAt + "/at".length() + 1));
+    }
+
+    /**
+     * Returns newly created Deadline task otherwise throw an appropriate error.
+     *
+     * @param userInput user input String
+     * @return Deadline task
+     * @throws DukeException exception thrown by Duke
+     */
+    public Deadline makeDeadlineTask(String userInput) throws DukeException {
+        int indexOfBy = userInput.indexOf("/by");
+        int endIndexForDeadline = "deadline".length();
+        boolean doesByExist = indexOfBy != -1;
+        boolean doesDateExist = userInput.length() > indexOfBy + "/by".length() + 1;
+        boolean doesDeadlineDescriptionExist = indexOfBy != endIndexForDeadline + 1;
+
+        if (!doesByExist) {
+            throw new DukeException("OOPS!!! You are missing /by before your date");
+        }
+        if (!doesDateExist) {
+            throw new DukeException("OOPS!!! The date of a deadline cannot be empty.");
+        }
+
+        if (!doesDeadlineDescriptionExist) {
+            throw new DukeException("OOPS!!! The description of a deadline cannot be empty.");
+        }
+        String stringDate = userInput.substring(indexOfBy + "/by".length() + 1);
+        return new Deadline(userInput.substring("deadline".length() + 1, indexOfBy - 1), stringDate);
+
+    }
+
+    /**
+     * Returns the number of tasks.
+     *
+     * @return number of tasks
+     */
+    public int noOfTasks() {
+        return tasks.size();
+    }
+
+    /**
+     * Returns encoded string list for saving.
```

Comment:
Same for here. Maybe you can try typing /** on top of the functions and press enter and intellij will help you to automatically add the necessary parameter in the javadoc